### PR TITLE
Rename Shared Blocks to Reusable Blocks

### PR DESCRIPTION
**Pull request by noisysocks**
This is the first half of https://github.com/WordPress/gutenberg/issues/7887.

Renames all 'shared block' references to 'reusable block', and changes the copy of the associated UI controls to match what was suggested in https://github.com/WordPress/gutenberg/issues/7887#issuecomment-407390960.

To test:
1. Ensure tests pass
2. Click around in Gutenberg and look for any regressions

### CODE REVIEW

**Comment by karmatosed** [comment]
Can we go with Reusable? Ideally saying 'Save this block to reuse' as the call? Apologies but I think this would work better going forward @noisysocks.

**Comment by noisysocks** [comment]
I've updated this to use 'reusable', changed the copy of the UI controls to match what Kjell suggested in https://github.com/WordPress/gutenberg/issues/7887#issuecomment-407390960, and added deprecated aliases for the public selectors, actions and block API.

File: editor/components/block-settings-menu/reusable-block-delete-button.js [hunk 1]
```diff
@@ -9,54 +9,54 @@ import { noop } from 'lodash';
...
 			className="editor-block-settings-menu__control"
 			icon="no"
-			disabled={ sharedBlock.isTemporary }
-			onClick={ () => onDelete( sharedBlock.id ) }
+			disabled={ reusableBlock.isTemporary }
+			onClick={ () => onDelete( reusableBlock.id ) }
 			role={ itemsRole }
 		>
-			{ __( 'Delete Shared Block' ) }
+			{ __( 'Remove from Reusable Blocks' ) }
```
Comment (youknowriad) [inline editor/components/block-settings-menu/reusable-block-delete-button.js]: This one is a bit confusing to me. I understand better "Remove reusable block" but I can see how the ambiguity there too.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by noisysocks** [comment]
Gonna merge this to get ahead of merge conflicts 🙂 – can change the copy later if @karmatosed isn't OK with it.

**Comment by aduth** [comment]
Any deprecations should be added to `docs/reference/deprecated.md`, which generates this page:

https://wordpress.org/gutenberg/handbook/reference/deprecated/

The functions deprecated here do not appear there and thus plugin authors who reference this page would not have the advance notice to have updated their usage.

File: editor/store/actions.js [hunk 2]
```diff
@@ -766,3 +767,63 @@ export function unregisterToken( name ) {
...
+	deprecated( 'updateSharedBlockTitle', {
+		alternative: 'updateReusableBlockTitle',
+		version: '3.6',
+		plugin: 'Gutenberg',
+	} );
+
+	return updateReusableBlockTitle( id, title );
+}
+
+export function convertBlockToSaved( clientId ) {
```
Comment (aduth) [inline editor/store/actions.js]: This function was called `convertBlockToShared`, not `convertBlockToSaved`.

Comment (noisysocks) [reply editor/store/actions.js]: 🤦‍♂️ good catch

**Pull request merged**